src/lib.rs

```diff
@@ -313,6 +313,7 @@ pub type SchemaValue = Vec<u8>;
 /// Represents operation written to the database.
 #[cfg_attr(feature = "arbitrary", derive(proptest_derive::Arbitrary))]
 #[derive(Debug, PartialEq, Eq, Hash, Clone)]
+// TODO: Do we want "generic" operation, with 2 options: S::Value and SchemaValue?
```

Comment:
Discussion point

Reply:
I think I'd like to see a `S::Value` variant of `Operation`, that'd probably make it easier to reason about the `struct` as well as safer. I'm not sure if the generic parameter is known in all situations where we currently deal with `Operation`, though.

For now `SchemaValue` is totally fine, but this is a good idea IMHO.

Reply:
I've created https://github.com/Sovereign-Labs/rockbound/issues/6 so  we can follow up later on it.